pkgs/development/python-modules/namex/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, wheel
+}:
+
+buildPythonPackage rec {
+  pname = "namex";
+  version = "0.0.7";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-hLplvE0ivZCePSa/L/tLlSm2CMs/mkM293awQgTO1ps=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  pythonImportsCheck = [ "namex" ];
+
+  # This packages has no tests.
+  doCheck = false;
```

Comment:
pytestCheckHook will run successfully with an empty test suite. why not do that?

Reply:
I fear not. The hook crashes when there is no test:
```
Check whether the following modules can be imported: namex
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Running phase: pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.11.7, pytest-7.4.3, pluggy-1.3.0
rootdir: /build/namex-0.0.7
collecting ... ^Mcollected 0 items                                                              

============================ no tests ran in 0.00s =============================
/nix/store/v5irq7wvkr7kih0hhnch5nnv2dcq8c4f-stdenv-linux/setup: line 1557: pop_var_context: head of shell_variables not a function context
```

Reply:
You will find many examples of python packages explicitly disabling `doCheck` when there are no tests:
```
❯ rg "doCheck = false;" pkgs/development/python-modules -c | wc -l
2349
```

Reply:
oops, you're right! i think that must be setup.py that has that behavior